src/Kunstmaan/NodeBundle/Helper/Services/PageCreatorService.php

```diff
@@ -95,118 +148,214 @@ public function setContainer(ContainerInterface $container = null)
      *
      * @throws \InvalidArgumentException
      */
-    public function createPage(HasNodeInterface $pageTypeInstance, array $translations, array $options = array())
+    public function createPage(HasNodeInterface $pageTypeInstance, array $translations, array $options = [])
     {
-        if (is_null($options)) {
-            $options = array();
-        }
+        $this->pageTypeInstance = $pageTypeInstance;
+        $this->options = (is_null($options)) ? [] : $options;
+        $this->throwExceptionIfNoTranslations($translations);
+        $this->seo = $this->getSeo();
+        $this->creator = $this->getCreator();
+        $this->nodeRepo = $this->getNodeRepo();
+        $this->parent = $this->getParent();
+        $this->translationsRepo = $this->getTranslationsRepo();
+
+        $firstTranslation = array_shift($translations);
+        $this->handleTranslation($firstTranslation, true);
 
-        if (is_null($translations) || (count($translations) == 0)) {
-            throw new \InvalidArgumentException('There has to be at least 1 translation in the translations array');
+        foreach ($translations as $translation) {
+            $this->handleTranslation($translation);
         }
 
-        $em = $this->entityManager;
+        $this->aclPermissionCreatorService->createPermission($this->rootNode);
+        return $this->rootNode;
+    }
 
+    /**
+     * @param array $translations
+     */
+    private function throwExceptionIfNoTranslations(array $translations)
+    {
+        if (is_null($translations) || count($translations) == 0) {
```

Comment:
```is_null``` is not necessary if argument is array type

Reply:
neither it is! That code was already there though! ;-) i'll also fix that